### DISCUSSION TITLE
Is there a way to block WebView2 from launching Windows Network Diagnostics window?

### DISCUSSION THREAD

**Discussion by stevealex1** [MicrosoftEdge/WebView2Feedback#2521] [Q&A]
In my WPF application with WebView2 control, it is required that any interaction (like clicking links) inside the page does not launch any external windows (i.e. any windows that is not part of the application).

In case a page the WebView2 is loading is not accessible or a timeout occurs, ERR_TIMED_OUT page is displayed with a link to run Windows Network Diagnostics as shown below.

![image](https://user-images.githubusercontent.com/102203144/173363567-23bcc867-b3a6-48c7-8b0f-99460afd45e3.png)

When clicking "Running Windows Network Diagnostics" link, it launches msdt.exe.

_Is there a way to prevent WebView2 from launching other applications and windows like msdt.exe?_

One suggested way was to define custom response pages that navigates to different error pages based on different error response status after the [NavigationCompleted Event](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.navigationcompleted?view=webview2-dotnet-1.0.1210.39).

_In this solution, can you get detailed information of the error? Are all HTTP error codes covered by the error codes obtained in NavigationCompleted event?_

**Comment by victorthoang** [comment]
hello @stevealex1 

Are you still looking for an answer to your question? If you still need a response, we can turn this into an Issue with a Question tag (which is typically how we address questions and bugs.

The discussion thread might be more useful for off topic questions or non-issue related inquiries!

Let me know if you'd like me to create an issue from this discussion, or if you'd prefer to add it. Thanks!
